node/src/node_configurator/node_configurator_standard.rs

```diff
@@ -1351,10 +1413,45 @@ mod tests {
 
         assert_eq!(
             result,
-            Err(ConfiguratorError::required(
-                "neighborhood-mode",
-                "Node cannot run as --neighborhood-mode zero-hop if --neighbors is specified"
-            ))
+            Ok(NeighborhoodConfig {
+                mode: NeighborhoodMode::ZeroHop
+            })
+        )
+    }
```

Comment:
I think there should be an assertion here that the PersistentConfiguration was used with the password to set the past-neighbors value. Or maybe this test is now useless duplication.